pkg/handlers/default.go

```diff
@@ -49,3 +56,68 @@ func KillContainer(pid int) registry.HandlerFunc {
 		return registry.HandlerResultErrno(unix.EPERM)
 	}
 }
+
+func FreezeContainer(pid int) registry.HandlerFunc {
+	re := regexp.MustCompile(`^(\d*):(freezer)?:(/..*)$`)
```

Comment:
What are we matching with this regex? starts with 0 or more numbers, followed by `:`, then `freezer` might be there or not, then `:` and something that starts with `/` and has at least one char (..*)?

Isn't there something simpler we can do, like find the index of the second instance of `:`, and take the substrig from there until the end? Is `:` a char that can appear in the fist two fields?

Reply:
I will add a comment:
```
	// Matches lines in /proc/$pid/cgroup for
	// - either the freezer cgroup (starting with '%d:freezer:'), or
	// - the unified hierarchy (starting with '0::')
```
So it should work on systems using cgroup-v1, cgroup-v2 or hybrid cgroups.

I want to match at least one character after `/` to make sure it is not the root cgroup (`/`) because that would freeze all processes on the node.

Reply:
But IIUC `:` is not a valid char for the first field nor the second field, using cgroups v1, v2 or hybrid. Right?

Why not use just the index of the second `:` and just slice it from there? It is way more readable and maintainable, etc. Or is there some edge cases to handle that makes this not so readable?


Also, the regex has several groups today, and we only use one. I think the `(\d)*` can probably just be `\d*`, without grouping, and maybe the second group can be avoided too. But this only applies IF we are using the regex, I personally don't like them and prefer to avoid them if the alternatives are more readable :)